From 058f3473e61ecfbd813d4bbd82aa3843d69f68e7 Mon Sep 17 00:00:00 2001
From: Franz-Josef Haider <frajo@frajo.fi>
Date: Tue, 30 Jun 2026 20:19:33 +0300
Subject: [PATCH 2/7] Revert "Be more explicit about using host hwbinder"

This reverts commit c102fee0dd8bc9ec15e88909eb35c7d0da72ae77.
---
 Binder.cpp                    |  7 -------
 BpHwBinder.cpp                |  5 -----
 Debug.cpp                     |  3 +--
 IInterface.cpp                | 11 -----------
 Parcel.cpp                    |  9 ++-------
 ProcessState.cpp              | 10 +++++-----
 include/hwbinder/BpHwBinder.h |  1 -
 include/hwbinder/IBinder.h    |  1 -
 include/hwbinder/IInterface.h |  1 -
 include/hwbinder/Parcel.h     |  1 -
 10 files changed, 8 insertions(+), 41 deletions(-)

diff --git a/Binder.cpp b/Binder.cpp
index 33b0a32..9edd27b 100644
--- a/Binder.cpp
+++ b/Binder.cpp
@@ -51,13 +51,6 @@ BpHwBinder* IBinder::remoteBinder()
     return nullptr;
 }
 
-bool IBinder::isHostHwBinder()
-{
-    auto binder = remoteBinder();
-    if (binder == nullptr) return false;
-    return binder->isHostHwBinder();
-}
-
 bool IBinder::checkSubclass(const void* /*subclassID*/) const
 {
     return false;
diff --git a/BpHwBinder.cpp b/BpHwBinder.cpp
index 9c449b0..f479f64 100644
--- a/BpHwBinder.cpp
+++ b/BpHwBinder.cpp
@@ -266,11 +266,6 @@ BpHwBinder* BpHwBinder::remoteBinder()
     return this;
 }
 
-bool BpHwBinder::isHostHwBinder()
-{
-    return mIsHostHwBinder;
-}
-
 BpHwBinder::~BpHwBinder()
 {
     ALOGV("Destroying BpHwBinder %p handle %d\n", this, mHandle);
diff --git a/Debug.cpp b/Debug.cpp
index 540bb85..4464032 100644
--- a/Debug.cpp
+++ b/Debug.cpp
@@ -301,8 +301,7 @@ void printHexData(int32_t indent, const void *buf, size_t length,
 }
 
 ssize_t getHWBinderKernelReferences(size_t count, uintptr_t* buf) {
-    const bool useHostHwBinder = false;
-    sp<ProcessState> proc = ProcessState::selfOrNull(useHostHwBinder);
+    sp<ProcessState> proc = ProcessState::selfOrNull();
     if (proc.get() == nullptr) {
         return 0;
     }
diff --git a/IInterface.cpp b/IInterface.cpp
index 76aaa83..ca8ad93 100644
--- a/IInterface.cpp
+++ b/IInterface.cpp
@@ -17,7 +17,6 @@
 #define LOG_TAG "hw-IInterface"
 #include <utils/Log.h>
 #include <hwbinder/IInterface.h>
-#include <hwbinder/BpHwBinder.h>
 
 namespace android {
 namespace hardware {
@@ -45,16 +44,6 @@ sp<IBinder> IInterface::asBinder(const sp<IInterface>& iface)
     return iface->onAsBinder();
 }
 
-// static
-bool IInterface::isHostHwBinder(const IInterface* iface)
-{
-    if (iface == nullptr) return false;
-    auto remoteBinder = const_cast<IInterface*>(iface)->onAsBinder()->remoteBinder();
-    if (remoteBinder == nullptr) return false;
-
-    return remoteBinder->isHostHwBinder();
-}
-
 
 // ---------------------------------------------------------------------------
 
diff --git a/Parcel.cpp b/Parcel.cpp
index 1a2f3be..5c5c5e9 100644
--- a/Parcel.cpp
+++ b/Parcel.cpp
@@ -250,14 +250,8 @@ status_t unflatten_binder(const sp<ProcessState>& proc,
 // ---------------------------------------------------------------------------
 
 Parcel::Parcel()
-    : Parcel(false)
 {
-}
-
-Parcel::Parcel(bool useHostHwBinder)
-{
-    LOG_ALLOC("Parcel %p: constructing, useHostHwBinder=%d", this, useHostHwBinder);
-    mIsHost = useHostHwBinder;
+    LOG_ALLOC("Parcel %p: constructing", this);
     initState();
 }
 
@@ -2008,6 +2002,7 @@ void Parcel::initState()
     mAllowFds = true;
     mOwner = nullptr;
     clearCache();
+    mIsHost = false;
 
     // racing multiple init leads only to multiple identical write
     if (gMaxFds == 0) {
diff --git a/ProcessState.cpp b/ProcessState.cpp
index 1c30528..1a0c885 100644
--- a/ProcessState.cpp
+++ b/ProcessState.cpp
@@ -49,20 +49,20 @@ namespace hardware {
 class PoolThread : public Thread
 {
 public:
-    explicit PoolThread(bool isMain, bool isHostHwBinder)
-        : mIsMain(isMain), mIsHostHwBinder(isHostHwBinder)
+    explicit PoolThread(bool isMain)
+        : mIsMain(isMain)
     {
     }
 
 protected:
     virtual bool threadLoop()
     {
-        IPCThreadState::self(mIsHostHwBinder)->joinThreadPool(mIsMain);
+        IPCThreadState::self()->joinThreadPool(mIsMain);
+        IPCThreadState::selfForHost()->joinThreadPool(mIsMain);
         return false;
     }
 
     const bool mIsMain;
-    const bool mIsHostHwBinder;
 };
 
 sp<ProcessState> ProcessState::self()
@@ -386,7 +386,7 @@ void ProcessState::spawnPooledThread(bool isMain)
     if (mThreadPoolStarted) {
         String8 name = makeBinderThreadName();
         ALOGV("Spawning new pooled thread, name=%s\n", name.string());
-        sp<Thread> t = new PoolThread(isMain, mIsHost);
+        sp<Thread> t = new PoolThread(isMain);
         t->run(name.string());
     }
 }
diff --git a/include/hwbinder/BpHwBinder.h b/include/hwbinder/BpHwBinder.h
index f3e9cc1..c58b638 100644
--- a/include/hwbinder/BpHwBinder.h
+++ b/include/hwbinder/BpHwBinder.h
@@ -54,7 +54,6 @@ public:
     virtual void        detachObject(const void* objectID);
 
     virtual BpHwBinder*   remoteBinder();
-            bool        isHostHwBinder();
 
             void        sendObituary();
                         // This refcount includes:
diff --git a/include/hwbinder/IBinder.h b/include/hwbinder/IBinder.h
index 7690e38..bc1f733 100644
--- a/include/hwbinder/IBinder.h
+++ b/include/hwbinder/IBinder.h
@@ -134,7 +134,6 @@ public:
 
     virtual BHwBinder*        localBinder();
     virtual BpHwBinder*       remoteBinder();
-    virtual bool              isHostHwBinder();
 
 protected:
     virtual          ~IBinder();
diff --git a/include/hwbinder/IInterface.h b/include/hwbinder/IInterface.h
index 8484277..7e13957 100644
--- a/include/hwbinder/IInterface.h
+++ b/include/hwbinder/IInterface.h
@@ -30,7 +30,6 @@ public:
             IInterface();
             static sp<IBinder>  asBinder(const IInterface*);
             static sp<IBinder>  asBinder(const sp<IInterface>&);
-            static bool         isHostHwBinder(const IInterface* iface);
 protected:
     virtual                     ~IInterface();
     virtual IBinder*            onAsBinder() = 0;
diff --git a/include/hwbinder/Parcel.h b/include/hwbinder/Parcel.h
index fc2bbe2..409ae86 100644
--- a/include/hwbinder/Parcel.h
+++ b/include/hwbinder/Parcel.h
@@ -52,7 +52,6 @@ class Parcel {
 public:
 
                         Parcel();
-                        Parcel(bool useHostHwBinder);
                         ~Parcel();
 
     const uint8_t*      data() const;
-- 
2.54.0

